pkg/infrastructure/azure/azure.go

```diff
@@ -512,7 +512,7 @@ func (p *Provider) InfraReady(ctx context.Context, in clusterapi.InfraReadyInput
 
 	var lbBap *armnetwork.BackendAddressPool
 	var extLBFQDN string
-	if in.InstallConfig.Config.Publish == types.ExternalPublishingStrategy {
+	if isAPIServerPublic(in) {
```

Comment:
We already have this check implemented.
```suggestion
	if in.InstallConfig.Config.PublicAPI() {
```

Reply:
Yes, https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L616 works.